include/Server.hpp

```diff
@@ -54,6 +54,15 @@ class Server
   std::vector<std::unique_ptr<XdgView>> &getViews();
   wm::WindowTree &getActiveWindowTree();
 
+  XdgView *getFocusedView() const noexcept
+  {
+    if (outputManager.getActiveWorkspace()->getViews().empty())
+      return nullptr;
+    return outputManager.getActiveWorkspace()->getViews().front().get();
+  }
+    
+  wlr_surface *getFocusedSurface() const noexcept;
```

Comment:
Why is this function not implemented like the others in this file ?

Reply:
I thought it was likelly to change implementation, so I wanted to avoid having systematic compilation.